docs/design.html

```diff
@@ -525,18 +526,23 @@ <h4><a id="design_compactionconfig" href="#design_compactionconfig">Configuring
     The log cleaner is enabled by default. This will start the pool of cleaner threads.
     To enable log cleaning on a particular topic, add the log-specific property
     <pre class="brush: text;"> log.cleanup.policy=compact</pre>
-    
-    The <code>log.cleanup.policy</code> property is a broker configuration setting defined 
-    in the broker's <code>server.properties</code> file; it affects all of the topics 
-    in the cluster that do not have a configuration override in place as documented 
+
+    The <code>log.cleanup.policy</code> property is a broker configuration setting defined
+    in the broker's <code>server.properties</code> file; it affects all of the topics
+    in the cluster that do not have a configuration override in place as documented
     <a href="/documentation.html#brokerconfigs">here</a>.
-    
+
     The log cleaner can be configured to retain a minimum amount of the uncompacted "head" of the log. This is enabled by setting the compaction time lag.
     <pre class="brush: text;">  log.cleaner.min.compaction.lag.ms</pre>
 
     This can be used to prevent messages newer than a minimum message age from being subject to compaction. If not set, all log segments are eligible for compaction except for the last segment, i.e. the one currently
     being written to. The active segment will not be compacted even if all of its messages are older than the minimum compaction time lag.
-    
+
+    The log cleaner can be configured to ensure a max delay after which the uncompacted "head" of the log becomes eligible for log compaction.
+    <pre class="brush: text;">  log.cleaner.max.compaction.lag.ms</pre>
+
+    This can be used to prevent log with low produce rate from remaining ineligible for compaction for an unbound duration. If not set, logs that do not exceed min.cleanable.dirty.ratio are not compacted.
```

Comment:
How about:
... unbound**ed**. Note that this compaction deadline is not a hard guarantee. You will want to monitor the uncleanable partitions, max-compaction-delay, and num-compaction-exceeded-partition-count metrics.

(substitute the actual metric names :) )